links/gql_exec/lib/src/http.dart

```diff
@@ -21,16 +21,35 @@ class BaseHttpLinkHeaders extends ContextEntry {
 
 /// HTTP link Response Context
 @immutable
-class BaseHttpLinkResponseContext extends ContextEntry {
+class HttpLinkResponseContext extends ContextEntry {
   /// HTTP status code of the response
   final int statusCode;
 
-  const BaseHttpLinkResponseContext({
+  final Map<String, String>? _headers;
+
+  final Map<String, List<String>>? _rawHeaders;
+
+  /// HTTP headers of the response. if you need duplicated headers, use [rawHeaders]
+  Map<String, String>? get headers =>
+      _headers ??
+      _rawHeaders?.map((key, valueList) =>
+          MapEntry(key, valueList.isEmpty ? "" : valueList.last));
+
+  /// HTTP headers
+  Map<String, List<String>>? get rawHeaders =>
+      _rawHeaders ?? _headers?.map((key, value) => MapEntry(key, [value]));
+
+  const HttpLinkResponseContext({
     required this.statusCode,
-  });
+    Map<String, String>? headers,
+    Map<String, List<String>>? rawHeaders,
```

Comment:
Why do we need both of these header maps?

Reply:
http headers can be repeated. [package:http does not support this](https://github.com/dart-lang/http/issues/24), but dio does. See **https://www.rfc-editor.org/rfc/rfc2616.html** 

So the alternatives are:

- only adding the  Map<String, List<String>> headers, but this would be a breaking change, and also would make the API more awkward for use cases that don't need repeated headers
- not support repeated headers at all
- add both

Reply:
Huh TIL